packages/flutter/lib/src/cupertino/text_theme.dart

```diff
@@ -0,0 +1,181 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart' show Brightness;
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+
+// Values derived from https://developer.apple.com/design/resources/.
+const TextStyle _kDefaultLightTextStyle = TextStyle(
+  inherit: false,
+  fontFamily: '.SF Pro Text',
+  fontSize: 17.0,
+  letterSpacing: -0.41,
+  color: CupertinoColors.black,
+  decoration: TextDecoration.none,
+);
+
+// Values derived from https://developer.apple.com/design/resources/.
+const TextStyle _kDefaultDarkTextStyle = TextStyle(
+  inherit: false,
+  fontFamily: '.SF Pro Text',
+  fontSize: 17.0,
+  letterSpacing: -0.41,
+  color: CupertinoColors.white,
+  decoration: TextDecoration.none,
+);
+
+// Values derived from https://developer.apple.com/design/resources/.
+const TextStyle _kDefaultActionTextStyle = TextStyle(
+  inherit: false,
+  fontFamily: '.SF Pro Text',
+  fontSize: 17.0,
+  letterSpacing: -0.41,
+  color: CupertinoColors.activeBlue,
+  decoration: TextDecoration.none,
+);
+
+// Values derived from https://developer.apple.com/design/resources/.
+const TextStyle _kDefaultTabLabelTextStyle = TextStyle(
+  inherit: false,
+  fontFamily: '.SF Pro Text',
+  fontSize: 10.0,
+  letterSpacing: -0.24,
+  color: CupertinoColors.inactiveGray,
+);
+
+const TextStyle _kDefaultMiddleTitleLightTextStyle = TextStyle(
+  inherit: false,
+  fontFamily: '.SF Pro Text',
+  fontSize: 17.0,
+  fontWeight: FontWeight.w600,
+  letterSpacing: -0.41,
+  color: CupertinoColors.black,
+);
+
+const TextStyle _kDefaultMiddleTitleDarkTextStyle = TextStyle(
+  inherit: false,
+  fontFamily: '.SF Pro Text',
+  fontSize: 17.0,
+  fontWeight: FontWeight.w600,
+  letterSpacing: -0.41,
+  color: CupertinoColors.white,
+);
+
+const TextStyle _kDefaultLargeTitleLightTextStyle = TextStyle(
+  inherit: false,
+  fontFamily: '.SF Pro Display',
+  fontSize: 34.0,
+  fontWeight: FontWeight.w700,
+  letterSpacing: 0.41,
+  color: CupertinoColors.black,
+);
+
+const TextStyle _kDefaultLargeTitleDarkTextStyle = TextStyle(
+  inherit: false,
+  fontFamily: '.SF Pro Display',
+  fontSize: 34.0,
+  fontWeight: FontWeight.w700,
+  letterSpacing: 0.41,
+  color: CupertinoColors.white,
+);
+
+/// Cupertino typography theme in a [CupertinoThemeData].
+@immutable
+class CupertinoTextTheme extends Diagnosticable {
```

Comment:
If `CupertinoTextTheme` is a property of `CupertinoThemeData` then shouldn't we call it `CupertinoTextThemeData`?  By naming convention, it would seem that `CupertinoTextTheme` is an analog to `CupertinoTheme`, but that's not the case, right?

Reply:
Ha, good call. I just took it from the Material ThemeData's TextTheme but I agree that one's the actual 'weird' one. Changed to CupertinoTextThemeData.